src/literal.h

```diff
@@ -197,6 +197,40 @@ class Literal {
         WASM_UNREACHABLE("unexpected type");
     }
   }
+
+  static Literal makeFromMemory(void* p, Type type) {
```

Comment:
Maybe the body could be in `literal.cpp`? Doesn't seem like it needs to be super-fast.